tests/functional/object/mcg/test_bucket_policy.py

```diff
@@ -1293,6 +1298,261 @@ def test_supported_bucket_policy_operations(self, mcg_obj, bucket_factory):
+        test_bucket_name,
+        bucketclass_dict,
+    ):
+        """
+        This test verified that anonymous user cannot access the bucket after public access block settings were applied
+        Scenario:
+        1. Create a bucket (or use predefined bucket) and write file
+        2. Verify that anonymous user cannot list the bucket content and cannot copy to the bucket
+        3. Put "allow all" policy to the bucket
+        4. Verify that anonymous user can list the bucket content but still cannot copy to the bucket
+        5. Put Public Access Block with Block/Restrict=True to the bucket
+        6. Verify that anonymous user cannot list the bucket content again and still cannot copy to the bucket
+        7. Put Public Access Block with Block/Restrict=False to the bucket
+        8. Verify that anonymous user can list the bucket content again.
+        Args:
+            mcg_obj (obj): An object representing the current state of the MCG in the cluster
+            awscli_pod_session (pod): A pod running the AWSCLI tools
+            bucket_factory: Calling this fixture creates a new bucket(s)
+            test_bucket_name(str) If is not empty, use the bucket by this name in the test. If empty -- create bucket
+            bucketclass_dict(dict) Parameters for bucket creation, used only if test_bucket_name is empty
+
+        """
+
+        bucket_name = (
+            bucket_factory(1, bucketclass=bucketclass_dict)[0].name
+            if not test_bucket_name
+            else test_bucket_name
+        )
+
+        # Copy a file to the bucket
+        standard_test_obj_list = awscli_pod_session.exec_cmd_on_pod(
+            f"ls -A1 {AWSCLI_TEST_OBJ_DIR}"
+        ).split(" ")
+        file_name = standard_test_obj_list[0]
+        logger.info(f"Going to copy file {file_name} to the bucket {bucket_name}")
+
+        TestS3BucketPolicy.check_commands(
+            mcg_obj, awscli_pod_session, bucket_name, file_name, False
+        )
+
+        bucket_policy_generated = gen_bucket_policy(
+            user_list="*",
+            actions_list=["*"],
+            resources_list=[bucket_name, f'{bucket_name}/{"*"}'],
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        # Put bucket policy
+        logger.info(f"Putting bucket policy {bucket_policy} on bucket: {bucket_name}")
+        put_bucket_policy(mcg_obj, bucket_name, bucket_policy)
+
+        # Hardcoded sleep is needed because we lack a confirmation mechanism
+        # we could wait for - even the get-policy result has been observed to be
+        # unreliable in confirming whether the policy is actually taking effect
+        timeout = 120
+        logger.info(f"Waiting for {timeout} seconds for the policy to take effect")
+        time.sleep(timeout)
+
+        get_policy = get_bucket_policy(mcg_obj, bucket_name)
+        logger.info(
+            f"Got bucket policy: {get_policy['Policy']} on bucket {bucket_name}"
+        )
```

Comment:
sleep time is set to 2 mins and we are using it 3 times, script will wait for 6 mins for each run and since we are having 6 iterations, it will wait around 36 mins in total, Instead of hardcoding I would suggest to use TimeoutSampler method and validate get_bucket_policy for every 30 seconds